scripts/filter_advisories.py

```diff
@@ -0,0 +1,126 @@
+import csv
+import json
+import re
+import sys
+from datetime import datetime
+from typing import Any, Dict, List
+
+import semantic_version
+
+
+def to_snake_case(s: str) -> str:
+    """Convert a string to snake_case."""
+    return re.sub(r"\W|^(?=\d)", "_", s).lower()
+
+
+def parse_advisories(filename: str) -> List[Dict[str, Any]]:
+    """Load advisories from a JSON file."""
+    with open(filename, "r") as f:
+        return json.load(f)
+
+
+def filter_advisories(
+    advisories: List[Dict[str, Any]],
+    minimal_version: semantic_version.Version,
+    last_run_date: datetime,
+) -> List[Dict[str, Any]]:
+    """Filter advisories based on the minimal vulnerable version and publication date."""
+    filtered = []
+    for advisory in advisories:
+        published_at = datetime.fromisoformat(advisory.get("published_at", ""))
+        if published_at > last_run_date:
```

Comment:
I'm not sure about this logic here - is it assuming that anything that has been seen a previous run has been already seen/actioned?

I think a more useful approach would be to print all the advisories where the range spec still matches, and treat this as something that needs fixing immediately.

Reply:
On first run, the action will publish all advisories that meet the version. After that, it's looking for new advisories. Posting the same advisories to the channel over and over (we won't be able to fix them) will lead to ignoring the alerts.